.github/workflows/e2e-run.yml

```diff
@@ -134,7 +134,7 @@ jobs:
   reports:
     runs-on: ubuntu-latest
     needs: tests
-    if: ${{ !cancelled() && github.repository == 'kafbat/kafka-ui' }}
+    if: ${{ !cancelled() && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository) }}
```

Comment:
This won't work as this will always be `workflow_dispatch`, as it's the only trigger for this particular workflow. 
I'm not sure what's the best way to do this properly, but the way we currently do this is that we forward the original `event_name` via workflow `inputs` (see `backend_tests.yml` for an example)

Reply:
so we only want this Reports task only run when the workflow triggers by workflow_dispatch (manually)?

Reply:
No, we still have PRs which are not from forks.

`github.event_name` here always would be `workflow_dispatch`, so as we'd need an original trigger name, we need to either forward it the way I described or some other way.

Reply:
Oh if that is the concern my current changes already handle it because I put `OR`/ `||`operation so it will still run for internal `pull request`

`${{!cancelled() && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository) }}`

you can check my testing repository 
Testing [Internal PR](https://github.com/agungardiyanta/docker-sample/actions/runs/9562700738/job/26359619835) 
<img width="1439" alt="image" src="https://github.com/kafbat/kafka-ui/assets/45070976/061531f4-fdee-4f42-957c-0f48e55bb198">

Testing [workflow_dispatch](https://github.com/agungardiyanta/docker-sample/actions/runs/9562676574/job/26359552343)
<img width="1405" alt="image" src="https://github.com/kafbat/kafka-ui/assets/45070976/c07bd155-23e6-454e-b10d-6321b19cdf92">

Testing Forks [Outside PR](https://github.com/agungardiyanta/docker-sample/actions/runs/9513452373/job/26344324400) 
<img width="1440" alt="image" src="https://github.com/kafbat/kafka-ui/assets/45070976/96d0af4f-8315-4c42-bb70-bb543b1ff02b">

Reply:
Once again, the `github.event_name` var will _never_ be anything but `workflow_dispatch` in this case, no matter which workflow calls the current one, this renders your if condition invalid. You can verify this by calling the workflow from three other caller workflows and dumping the github context into logs. 

We need to forward the original event name from the caller workflow (or use any other mechanism if you're aware of ones).

Reply:
Hi, @Haarolean I did a little experiment in my repo to test what the `github.event_name` will be received by template workflow `e2e.yaml`, and it seems my workflow can get the original event_name without adding a new parameter CMIIW. 
[workflow_dispatch](https://github.com/agungardiyanta/docker-sample/blob/master/.github/workflows/manual.yml)
<img width="1440" alt="image" src="https://github.com/kafbat/kafka-ui/assets/45070976/b99ee13c-f208-4fe4-8423-905b7122d10f">

[pull request](https://github.com/agungardiyanta/docker-sample/blob/master/.github/workflows/pr.yaml)
<img width="1440" alt="image" src="https://github.com/kafbat/kafka-ui/assets/45070976/f21fc49b-79ab-447f-b537-f1177617d2fb">